bpf/include/bpf_log.h

```diff
@@ -48,4 +48,12 @@ enum bpf_loglevel {
         }                                                                                                              \
     } while (0)
 
+/* add this macro to get ip addr from ctx variable, include bpf_sock_addr or bpf_sock_ops, weird
+reason is direct access would not be print ipaddr when pass `&ctx->remote_ipv4` to bpf_trace_printk */
+#define DECLARE_VAR_IPV4(t, ctx_ip, name)                                                                              \
+    __u32 name = 0;                                                                                                    \
+    if ((int)(BPF_LOGTYPE_##t) == BPF_DEBUG_ON) {                                                                      \
```

Comment:
how to format ipv6 addr

Reply:
This macro seems not a must

Reply:
> how to format ipv6 addr
It may need to define another macro to get ipv6, then pass ipv6 pointer to bpf_trace_printk using format `%pI6c`.
#https://www.kernel.org/doc/html/v4.20/core-api/printk-formats.html#ipv6-addresses

Reply:
> This macro seems not a must

I consider using the macro to avoid defining ip variable when prog need to print the ipaddr info of bpf context, it could reduce code lines. 
Also when one of ebpf prog's debug is off, it need not to get ip info from bpf context.